packages/core-bridge/src/helpers/handles.rs

```diff
@@ -0,0 +1,149 @@
+use std::{
+    any,
+    cell::{Ref, RefCell},
+    sync::Arc,
+};
+
+use neon::{prelude::*, types::JsBox};
+
+use super::{BridgeError, BridgeResult, TryFromJs, TryIntoJs};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Opaque Handles are native structures that are sent into the JS side but without exposing
+/// their internal structure; i.e. they are only meant to be passed back to the Rust side
+/// on appropriate API calls.
+///
+/// Due to very different ownership models and usage patterns, and to avoid unnecessary
+/// complexity, Opaque Handles are split into two variants:
+///
+/// - `OpaqueOutboundHandle` - for sending an opaque handle that was just created by the Rust
+///                            side to the JS side (e.g. as the return value of an API call).
+/// - `OpaqueInboundHandle` - for accepting an opaque handle sent from the JS side to the Rust
+///                           side (e.g. as an argument to an API call).
+pub struct OpaqueOutboundHandle<T: MutableFinalize + 'static> {
+    inner: T,
+}
+
+impl<T: MutableFinalize + 'static> OpaqueOutboundHandle<T> {
+    // We late-wrap into Arc<Box<RefCell<>>> so that T can be send across threads without mutex.
```

Comment:
Where does the `Box` come from here? I don't actually see that in the code? An `Arc<Box<...>>` pretty much never makes sense really, since that's just 2x pointer indirection when 1 would do

Reply:
Additionally an `Arc<RefCell<>>` seems like something that doesn't make sense. `Arc` is only `Send + Sync` if its inner type is, but `RefCell` isn't, so this outer type can't be either. Seemingly this should just be an `Rc`?

Reply:
> Where does the Box come from here? I don't actually see that in the code? An Arc<Box<...>> pretty much never makes sense really, since that's just 2x pointer indirection when 1 would do

Indeed. Stale comment.

> `Arc<RefCell<>>` seems like something that doesn't make sense

Indeed. It is now `JsBox<Rc<Wrapper<T>>>`, where `Wrapper<T> = RefCell<Option<T>>`.